test/components/buttons/CartQuantity.test.tsx

```diff
@@ -0,0 +1,304 @@
+// import { useAppDispatch, useAppSelector } from '../../../src/redux/hooks/hooks';
+// import CartQuantity from '../../../src/components/buttons/CartQuantity';
+// import AllProvider from '../../Utils/AllProvider';
+// import cartReducer from '../../../src/redux/features/cartSlice';
+// vi.mock('../../../src/hooks/useToast');
+// vi.mock('../../../src/redux/hooks/hooks');
+
+// const mockStore = configureStore({
+// 	reducer: {
+// 		cart: cartReducer,
+// 	},
+// });
+
+// const customRender = ({ store = mockStore, ...renderOptions } = {}) => {
+// 	return render(<CartQuantity productId="234543" />, {
+// 		wrapper: AllProvider,
+// 		...renderOptions,
+// 	});
+// };
+// describe('CartQuantity Component', () => {
+// 	const showErrorMessageMock = vi.fn();
+// 	const dispatchMock = vi.fn();
+
+// 	beforeEach(() => {
+// 		(useToast as vi.Mock).mockReturnValue({
+// 			showErrorMessage: showErrorMessageMock,
+// 		});
+// 		(useAppDispatch as vi.Mock).mockReturnValue(dispatchMock);
+// 		(useAppSelector as vi.Mock).mockReturnValue({
+// 			carts: {
+// 				products: [{ id: '1', quantity: 3 }],
+// 			},
+// 		});
+// 	});
+
+// 	afterEach(() => {
+// 		vi.clearAllMocks();
+// 	});
+
+// 	test('should render initial quantity from state', () => {
+// 		customRender(<CartQuantity productId="1" />);
+
+// 		expect(screen.getByText('0')).toBeInTheDocument();
+// 	});
+
+// 	test('should call dispatch when increment button is clicked', async () => {
+// 		customRender(<CartQuantity productId="1" />);
+// 		const incrementButton = screen.getByText('+');
+
+// 		await fireEvent.click(incrementButton);
+
+// 		expect(dispatchMock).toHaveBeenCalledWith(expect.any(Function)); // check for dispatch call
+// 	});
+
+// 	// test('should call dispatch when decrement button is clicked', async () => {
+// 	// 	customRender(<CartQuantity productId="1" />);
+// 	// 	const decrementButton = screen.getByText('-');
+
+// 	// 	await fireEvent.click(decrementButton);
+
+// 	// 	expect(dispatchMock).toHaveBeenCalledWith(expect.any(Function)); // check for dispatch call
+// 	// });
+
+// 	test('should not decrement quantity below 1', async () => {
+// 		(useAppSelector as vi.Mock).mockReturnValueOnce({
+// 			carts: {
+// 				products: [{ id: '1', quantity: 1 }],
+// 			},
+// 		});
+
+// 		customRender(<CartQuantity productId="1" />);
+// 		const decrementButton = screen.getByText('-');
+
+// 		await fireEvent.click(decrementButton);
+
+// 		expect(dispatchMock).not.toHaveBeenCalledWith(expect.any(Function)); // no dispatch call for decrement
+// 	});
+
+// 	// test('should show error message if dispatch fails', async () => {
+// 	// 	dispatchMock.mockRejectedValueOnce(new Error('Test error'));
+// 	// 	customRender(<CartQuantity productId="1" />);
+// 	// 	const incrementButton = screen.getByText('+');
+
+// 	// 	await fireEvent.click(incrementButton);
+
+// 	// 	expect(showErrorMessageMock).toHaveBeenCalledWith('Test error');
+// 	// });
```

Comment:
remove this code if they are not used anymore

Reply:
i remove it @Tuyisenge2